node/core/dispute-coordinator/src/initialized.rs

```diff
@@ -1036,6 +1036,13 @@ impl Initialized {
 				session,
 				"Dispute on candidate concluded with 'valid' result",
 			);
+			if new_state.has_own_vote() && new_state.own_approval_votes().is_none() {
```

Comment:
Approval votes are not the only way to vote "valid", there are also explicit valid votes.